test/JDBC/input/BABEL-3395-vu-verify.sql

```diff
@@ -0,0 +1,88 @@
+DECLARE @class NCHAR(30) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE (@class) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'ANIKAIT AGRAWAL'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'Anikait Agrawal '
+SELECT '[' + @class + ']' WHERE lower(@class) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE lower(@class) = N'anikait agrawal'
+SELECT '[' + @class + ']' WHERE lower(@class) = N'Anikait Agrawal '
+GO
+
+DECLARE @class CHAR(30) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE (@class) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'ANIKAIT AGRAWAL'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'Anikait Agrawal '
+SELECT '[' + @class + ']' WHERE lower(@class) = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE lower(@class) = N'anikait agrawal'
+SELECT '[' + @class + ']' WHERE lower(@class) = N'Anikait Agrawal '
+GO
+
+DECLARE @class NCHAR = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'ANIKAIT AGRAWAL'
+SELECT '[' + @class + ']' WHERE lower(@class) = N'anikait agrawal'
+GO
+
+DECLARE @class CHAR = N'Anikait Agrawal'
+SELECT '[' + @class + ']' WHERE upper(@class) = N'ANIKAIT AGRAWAL'
+SELECT '[' + @class + ']' WHERE lower(@class) = N'anikait agrawal'
+GO
+
+-- different result from TSQL, should be fixed under BABEL-4807
+declare @string1 varchar(30) = N'比尔·拉' COLLATE chinese_prc_ci_as
+SELECT UPPER(@string1)
+SELECT UPPER(@string1) COLLATE chinese_prc_ci_as
+SELECT LOWER(@string1)
+SELECT LOWER(@string1) COLLATE chinese_prc_ci_as
+go
+
+declare @string1 char(30) = N'比尔·拉' COLLATE chinese_prc_ci_as
+SELECT UPPER(@string1)
+SELECT UPPER(@string1) COLLATE chinese_prc_ci_as
+SELECT LOWER(@string1)
+SELECT LOWER(@string1) COLLATE chinese_prc_ci_as
+go
+
+SELECT UPPER(NULL)
+SELECT LOWER(NULL)
+GO
+
+DECLARE @inputString BINARY(30) = 0x4142434445
+SELECT UPPER(@inputString)
```

Comment:
Add following tests for BINARY to check if NULL characters appended for binary datatype is getting retained or not
```
SELECT CONVERT(VARBINARY(50), UPPER(@inputString));
SELECT CONVERT(VARBINARY(50), LOWER(@inputString));
```

Add following tests for CHAR and NCHAR to check if padding of CHAR and NCHAR datatype are getting retained or not.
```
SELECT '|' + UPPER(@inputString) + '|'`
SELECT '|' + LOWER(@inputString) + '|'`
```

Reply:
Done